spec/jobs/irs_attempts_events_batch_job_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe IrsAttemptsEventsBatchJob, type: :job do
+  describe '#perform' do
+    context 'IRS attempts API is enabled' do
+      let(:start_time) { Time.new(2020, 1, 1, 12, 0, 0, 'UTC') }
+      let(:private_key) { OpenSSL::PKey::RSA.new(4096) }
+      let(:events) do
+        [
+          {
+            event_key: 'key1',
+            jwe: 'some_event_data_encrypted_with_jwe',
+            timestamp: start_time + 10.minutes,
+          },
+          {
+            event_key: 'key2',
+            jwe: 'some_other_event_data_encrypted_with_jwe',
+            timestamp: start_time + 15.minutes,
+          },
+        ]
+      end
+
+      before do
+        allow(IdentityConfig.store).to receive(:irs_attempt_api_enabled).and_return(true)
+        allow(IdentityConfig.store).to receive(:irs_attempt_api_public_key).
+          and_return(Base64.strict_encode64(private_key.public_key.to_der))
+
+        Dir.mktmpdir do |dir|
+          @dir_path = dir
+        end
```

Comment:
this tempdir will be cleaned up inside the before block. We want this in an `around` block instead

```suggestion
      around do |ex|
        Dir.mktmpdir do |dir|
          @dir_path = dir
          ex.run
        end
      end

      before do
        allow(IdentityConfig.store).to receive(:irs_attempt_api_enabled).and_return(true)
        allow(IdentityConfig.store).to receive(:irs_attempt_api_public_key).
          and_return(Base64.strict_encode64(private_key.public_key.to_der))
```